modules/standard/FileSystem.chpl

```diff
@@ -861,6 +869,10 @@ proc isLink(out error:syserr, name: string): bool {
   extern proc chpl_fs_is_link(ref result:c_int, name: c_string): syserr;
 
   var ret:c_int;
+  var doesExist = exists(error, name);
```

Comment:
I can see the value of returning false for ENOTDIR (which we can check for in the error returned by exists), but the other errors seem to me to provide information the user likely wants.